BackEnd/src/test/kotlin/quest/marketstack/tradingapp/opentrades/controller/OpenOpenTradeControllerTest.kt

```diff
@@ -0,0 +1,348 @@
+package quest.marketstack.tradingapp.opentrades.controller
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.hamcrest.Matchers.hasSize
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.MediaType
+import org.springframework.test.annotation.DirtiesContext
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.get
+import org.springframework.test.web.servlet.post
+import quest.marketstack.tradingapp.model.TradeExec
+import java.time.LocalDate
+import java.time.LocalTime
+
+@ActiveProfiles("test")
+@SpringBootTest
+@AutoConfigureMockMvc
+class OpenOpenTradeControllerTest
```

Comment:
This whole test class is like half functional/integration test, and half unit test, so it's not really doing the correct thing. I mentioned in the previous comment that I wanted you to delete all the test classes you have and rewrite them based on the example I provided. That definitely still applies to this and all the other test classes, but specifically this one I wanted to call out that this isn't the best way of doing things. Start with just writing unit tests for the controller the way I told you, and then once you have all the unit tests done for everything, let's get on a call and I can walk you through functional tests